filegenerator/file_generator.go

```diff
@@ -0,0 +1,361 @@
+	dateSet := make(map[string]bool)
+	for _, dir := range packages {
+		for _, tests := range dir {
+			for date := range tests {
+				dateSet[date] = true
+			}
+		}
+	}
+	for date := range dateSet {
+		dates = append(dates, date)
+	}
+	return dates
+}
+
+// GenerateGeometricMeanJSONFile creates a JSON file that has the geometric mean of
+// the results from every benchmark test in a package for every package.
+func GenerateGeometricMeanJSONFile(packages BenchPackages, dirs []string) {
+	deployRoot := os.Getenv(deployRootEnv)
+	dates := getDatesFromPackages(packages)
+	packageToGeometricMean := make(map[string][]GeometricMeanData)
+	for _, dir := range dirs {
+		packageToGeometricMean[dir] = make([]GeometricMeanData, 0)
+		for _, date := range dates {
+			vectors := make(map[string][]float64)
+			for _, test := range packages[dir] {
+				vectors["N"] = append(vectors["N"], float64(test[date].N))
+				vectors["A"] = append(vectors["A"], float64(test[date].A))
+				vectors["B"] = append(vectors["B"], float64(test[date].B))
+				vectors["M"] = append(vectors["M"], float64(test[date].M))
+			}
+			nMean := GetGeometricMean(vectors["N"])
+			aMean := GetGeometricMean(vectors["A"])
+			bMean := GetGeometricMean(vectors["B"])
+			mMean := GetGeometricMean(vectors["M"])
+			packageToGeometricMean[dir] = append(packageToGeometricMean[dir],
+				GeometricMeanData{NMean: nMean, AMean: aMean, BMean: bMean, MMean: mMean, Date: date})
+		}
+	}
+	fileName := filepath.Join(deployRoot, "/geometric_means.json")
+	geometricMeansJSON, err := json.Marshal(&packageToGeometricMean)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = ioutil.WriteFile(fileName, geometricMeansJSON, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// GetGeometricMean returns the geometric mean of a vector.
+func GetGeometricMean(vector []float64) float64 {
+	epsilon := .0001
+	sum := 0.0
+	size := 0
+	/// Use law of logs to avoid overflows when multiplying entire vector.
+	for _, num := range vector {
+		if num > epsilon {
+			sum += math.Log(num)
+			size++
+		}
+	}
+	if size == 0 {
+		return 0
+	}
+	return math.Exp(sum / float64(size))
+}
+
+func getFormattedDate() string {
+	timeData := strings.Split(strings.Split(time.Now().String(), " ")[0], "-")
+	timeFormatted := timeData[2] + "-" + timeData[1] + "-" + timeData[0]
+	return timeFormatted
+}
+
+// This function creates all the necessary directories and subdirectories
+// for a sampleBench folder that mimics what the Marc folders look like
+func prepDirectories(dirs []string, date string) {
+	benchSamples := os.Getenv(benchSamplesEnv)
+	_ = os.MkdirAll(filepath.Join(benchSamples, date, "cockroach"), 0755)
+	for _, dir := range dirs {
+		_ = os.MkdirAll(filepath.Join(benchSamples, date, "cockroach", dir), 0755)
+	}
+}
+
+// TODO (Will Haack) Potentially remove this method if results are already being
+// generated and saved correctly.
+func GenerateBenchmarkResults(dirs []string) {
+	benchSamples := os.Getenv(benchSamplesEnv)
```

Comment:
removed cockroachRoot since we're not using this script to generate the benchmark tests